src/substrait/textplan/README.md

```diff
@@ -0,0 +1,208 @@
+| interval_day         | duration specified in days, hours, minutes, seconds, and microseconds    |
+| timestamp_tz         | timestamp with a specified timezone                                      |
+| uuid                 | 128-bit UUID                                                             |
+|                      |                                                                          |
+| varchar\<L>          | a variable sized character sequence of up to length L                    |
+| fixedchar\<L>        | a character sequence of fixed length L                                   |
+| fixedbinary\<L>      | a fixed length binary sequence of length L                               |
+| decimal\<P,S>        | a decimal value with precision P and scale S                             |
+|                      |                                                                          |
+| struct<T1,T2,...,TN> | a structure consisting of N specified types (may include nested structs) |  
+| list\<T>             | a list of type T where T can be any type (including a list)              | 
+| map\<K,V>            | a map with keys of type K and values of type V                           |
+
+Note that user defined types are not yet supported in text plans (but will be at a future date).
+
+#### Example Types
+
+|                                        | Explanation                                                         |
+|----------------------------------------|---------------------------------------------------------------------|
+| i64\?                                  | optional int64                                                      |
+| varchar\<5>                            | a string with up to five characters                                 |
+| struct\<string,i8,i32?,timestamp_tz>   | a structure with a string, two ints, one optional, and a timestamp  |
+| struct\<string,struct\<string,string>> | a structure containing a string and a substructure with two strings |
+
+## Literals
+
+Types are important in Substrait plans as the choice of type controls which implementation of a given function should
+be used.  In the text representation of plans we label literals with their types to make it clear which function should
+be invoked within the plan. The type of a literal is appended to the literal behind an underscore (_).  Typed literals 
+may not be optional.
+
+### Integer Literals
+
+There are four different integer widths.  While other languages provide for binary, octal, and hexadecimal
+representations of their integer literals (as well as spacing separators) human readability of literals isn't a current
+requirement of the language and so are omitted for brevity.
+
+#### Example Integer Literals
+```
+123_i8
+0_i16
+123_i32
+1234_i64
+```
+
+### Floating Point Literals
+
+Floating point literals are represented with by a decimal number followed by an optional exponent.   
```

Comment:
Does the exponent always need a sign char?  I recently found out that 10e7 is not a numeric YAML literal but 10e+7 is :)

Reply:
Ah, that's to keep the hexadecimal literals distinct from the floating point ones.  It's not an issue if we don't have hexadecimal literals but might as well make the distinction in case we decide to upgrade later on.